Source/AwsCommonRuntimeKit/crt/Utilities.swift

```diff
@@ -272,3 +272,37 @@ func withByteCursorFromStrings<Result>(
         }
     }
 }
+
+public func scan<S: Sequence, U>(_ seq: S, _ initial: U, _ combine: (U, S.Element) -> U) -> [U] {
+    var result: [U] = []
+    result.reserveCapacity(seq.underestimatedCount)
+    var runningResult = initial
+    for element in seq {
+    runningResult = combine(runningResult, element)
+    result.append(runningResult)
+    }
+    return result
+}
+
+/// Note: this function and associated scan function above are copied verbatim from
+/// swift standard lib where its private. 
+public func withArrayOfCStrings<R>(
```

Comment:
These functions should be internal to our module and not public. Also, if performance is not a concern, we should just implement the simpler version from https://oleb.net/blog/2016/10/swift-array-of-c-strings/.

Reply:
ugh, i didnt really understand how swift works. updated to a much simpler version.